index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+          |gamepadHapticActuator:GamepadHapticActuator|</dfn> representing a
+          {{Gamepad}}'s primary vibration actuator is constructed by
+          performing the following steps:
+        </p>
+        <ol>
+          <li>Let |gamepadHapticActuator:GamepadHapticActuator| be a newly
+          created {{GamepadHapticActuator}} instance.
+          </li>
+          <li>Let `supportedEffectsList` be an empty list.
+          </li>
+          <li>For each enum value |type:GamepadHapticEffectType| of
+          {{GamepadHapticEffectType}}, if the [=user agent=] can send a
+          command to initiate effects of that type on that actuator, append
+          |type| to `supportedEffectsList`.
+          </li>
+          <li>Set
+          |gamepadHapticActuator|.{{GamepadHapticActuator/[[effects]]}} to
+          `supportedEffectsList`.
+          </li>
+        </ol>
+      </section>
+    </section>
+    <section>
+      <h3>
+        <dfn>GamepadHapticsResult</dfn> Enum
+      </h3>
+      <p>
+        An enum representing the result of a call to
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect()}} or
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/reset()}}.
+      </p>
+      <pre class="idl">
+        enum GamepadHapticsResult {
+          "complete",
+          "preempted"
+        };
+      </pre>
+      <dl data-dfn-for="GamepadHapticsResult">
+        <dt>
+          <dfn>complete</dfn>
+        </dt>
+        <dd>
+          <p>
+            The value the {{Promise}} returned by
+            {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect()}}
+            or {{GamepadHapticActuator}}.{{GamepadHapticActuator/reset()}}
+            will resolve to when the effect completes normally.
+          </p>
+        </dd>
+        <dt>
+          <dfn>preempted</dfn>
+        </dt>
+        <dd>
+          <p>
+            The value the {{Promise}} returned by
+            {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect()}}
+            or {{GamepadHapticActuator}}.{{GamepadHapticActuator/reset()}}
+            will resolve to when the operation does not complete because the
+            {{Document}} is `"hidden"` or another effect was started
+            on the same actuator.
+          </p>
+        </dd>
+      </dl>
+    </section>
+    <section>
+      <h3>
+        <dfn>GamepadHapticEffectType</dfn> Enum
+      </h3>
+      <p>
+        The effect type defines how the effect parameters are interpreted by
+        the actuator.
+      </p>
+      <p>
+        The [=user agent=] MAY modify the behavior of
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect}} to
+        extend support for new {{GamepadHapticEffectType}} values and
+        {{GamepadEffectParameter}} attributes. Implementations MUST NOT
+        modify the definition of a [=valid effect=] for effect types defined
+        in this specification.
+      </p>
```

Comment:
New  GamepadHapticEffectType should be standardized instead. Definitely don't want to encourage support for arbitrary new types! The second clause doesn't seem to be very useful to be honest. 

```suggestion
```